pkg/metallb/frr.go

```diff
@@ -0,0 +1,664 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	frrtypes "github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FrrConfigurationBuilder provides struct for the FrrConfiguration object containing connection to
+// the cluster and the FrrConfiguration definitions.
+type FrrConfigurationBuilder struct {
+	Definition *frrtypes.FRRConfiguration
+	Object     *frrtypes.FRRConfiguration
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// NewFrrConfigurationBuilder creates a new instance of FRRConfiguration.
+func NewFrrConfigurationBuilder(
+	apiClient *clients.Settings, name, nsname string) *FrrConfigurationBuilder {
+	glog.V(100).Infof(
+		"Initializing new Frrconfiguration structure with the following params: %s, %s",
+		name, nsname)
+
```

Comment:
Make sure to [attach the scheme](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/metallb/metallb.go#L35-L40) that way you can use the runtime client instead of using unstructured

Reply:
added